ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2

```diff
@@ -17,6 +17,12 @@ ethernet_interfaces:
 {%             set peer = connected_endpoint %}
 {%             for adapter_switch in adapter.switches %}
 {%                 if adapter_switch == inventory_hostname %}
+{%                     if adapter_settings.monitor_sessions is arista.avd.defined %}
+{%                         for monitor_session in adapter_settings.monitor_sessions %}
+{%                             do monitor_session.update({'interfaces': adapter.switch_ports}) %}
```

Comment:
I'm not following you on this one Claus. It's valid to define multiple switchports per switch via adapter.switch_ports, but they would all need to have the same monitor_session config (because we don't have a knob to determine which monitoring config should apply for which interface, they would all get the same config). That's why I thought it's ok to add the list of switchports to "interfaces"?

Reply:
In line 18 we run a for loop per switch entry and check if the switch is "us" in line 19, so we should only add the session for the interface relevant to this switch. Remember the template will run once per switch.
Even in case of multiple interfaces connected to the same switch (with multiple entries with same name in `adapter.switches` we will add them one-by-one to `monitor_session_data`.
So I suggest to change the key to `interface`.